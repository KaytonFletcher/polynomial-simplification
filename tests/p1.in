(x+2)^4+(-1)*(3x)*(2x-1)
