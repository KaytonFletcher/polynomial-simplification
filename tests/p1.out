(((x+2))^4+((-(1)*(3*x))*((2*x)-1)))
4
found times! 
distribute term within plus to other plus
distribute term within plus to other plus
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
1
2
found times! 
2
original list is empty 
3
found times! 
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
1
4
found times! 
Times with at least two terms. n1 and n3 -13
2
original list is empty 
2
found times! 
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distribute term within plus to other plus
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
distTerm
Term -> Term -> Term
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
1
3
found times! 
Times with at least two terms. n1 and n3 -31
2
original list is empty 
1
found times! 
Only one thing in times
1
2
found times! 
distTerm
Times in plus distTerm
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
Times with exactly two terms. n1: 2
distTerm
Term -> Term -> Term
distTerm
list being distributed is empty
Only one thing in times
2
original list is empty 
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with at least two terms
Plus with exactly two terms
1x^4+8x^3+18x^2+35x+16